optimize/qa/schema-integrity-tests/src/test/java/io/camunda/optimize/test/upgrade/UpgradeElasticsearchSchemaIT.java

```diff
@@ -52,30 +50,29 @@ protected void assertMigratedDatabaseIndicesMatchExpected() throws IOException {
     final Map<String, Map> newSettings = newDatabaseSchemaClient.getSettings();
     LOG.info("Actual settings size: {}, keys: {}", newSettings.size(), newSettings.keySet());
     assertThat(newSettings).isEqualTo(expectedSettings);
-    assertThat(newDatabaseSchemaClient.getMappings()).isEqualTo(expectedMappings);
+    assertThat(newDatabaseSchemaClient.getMappings())
+        .isEqualToComparingFieldByFieldRecursively(expectedMappings);
   }
 
   @Override
   protected void assertMigratedDatabaseAliasesMatchExpected() throws IOException {
     LOG.info(
         "Expected aliases size: {}, keys: {}", expectedAliases.size(), expectedAliases.keySet());
-    final Map<String, Set<AliasMetadata>> newAliases = newDatabaseSchemaClient.getAliases();
+    final Map<String, IndexAliases> newAliases = newDatabaseSchemaClient.getAliases();
     LOG.info("Actual aliases size: {}, keys: {}", newAliases.size(), newAliases.keySet());
-    assertThat(newAliases).isEqualTo(expectedAliases);
+    assertThat(newAliases).isEqualToComparingFieldByFieldRecursively(expectedAliases);
   }
 
   @Override
   protected void assertMigratedDatabaseTemplatesMatchExpected() throws IOException {
     LOG.info(
         "Expected templates size: {}, names: {}",
         expectedTemplates.size(),
-        expectedTemplates.stream().map(IndexTemplateMetadata::name).toList());
-    final List<IndexTemplateMetadata> newTemplates = newDatabaseSchemaClient.getTemplates();
-    LOG.info(
-        "Actual templates size: {}, names: {}",
-        newTemplates.size(),
-        newTemplates.stream().map(IndexTemplateMetadata::name).toList());
-    assertThat(newTemplates).containsExactlyInAnyOrderElementsOf(expectedTemplates);
+        expectedTemplates.keySet());
+    final Map<String, TemplateMapping> newTemplates = newDatabaseSchemaClient.getTemplates();
+    LOG.info("Actual templates size: {}, names: {}", newTemplates.size(), newTemplates.keySet());
+    assertThat(newTemplates.keySet())
+        .containsExactlyInAnyOrderElementsOf(expectedTemplates.keySet());
```

Comment:
by just comparing keyset, are we not effectively just comparing the names of the templates? Previously, I believe the actual templates themselves were being asserted on, it would be good if we can keep that behaviour

Reply:
fixed